src/antlr/QueryStructureVisitor.ts

```diff
@@ -339,6 +343,45 @@ export abstract class QueryStructureVisitor<Result> extends AbstractParseTreeVis
     return result;
   }
 
+  /**
+   * Process JOIN ... USING (...) columns
+   * @param ctx
+   */
+  visitJoinCriteriaUsing(ctx: JoinCriteriaUsingContext): Result {
+    const columns = ctx
+      .identifierList()
+      .identifierSeq()
+      .errorCapturingIdentifier()
+      .map(eci => common.stripQuote(eci.identifier()));
+
+    // columns shall be searched in last from relation and all previous ones
+    const size = this.currentRelation.relations.size;
+    if (size >= 2) {
+      let i = 0;
+      const foundLeftCol: Array<boolean> = [];
+      this.currentRelation.relations.forEach(r => {
```

Comment:
You can also use the indexed version of `forEach` to get rid of the local `i`.

Reply:
It is a Map and unfortunately we don't have index in `forEach()` there.